gateway/ln-gateway/src/lib.rs

```diff
@@ -163,22 +166,19 @@ impl Gateway {
 
         let mut gw = Self {
             lnrpc,
-            actors: Arc::new(RwLock::new(BTreeMap::new())),
+            clients: Arc::new(RwLock::new(BTreeMap::new())),
             scid_to_federation: Arc::new(RwLock::new(BTreeMap::new())),
             sender,
             receiver,
             client_builder,
             channel_id_generator: AtomicU64::new(INITIAL_SCID),
-            decoders: decoders.clone(),
-            module_gens: module_gens.clone(),
             lightning_mode: Some(lightning_mode),
             fees,
             gatewayd_db,
+            api,
         };
 
-        let (route_hints, node_pub_key) = gw.fetch_route_hints_and_node_pub_key().await?;
-        gw.load_actors(decoders, module_gens, route_hints, node_pub_key)
-            .await?;
+        gw.load_clients(task_group).await?;
```

Comment:
Might be nice to make the `clients` a newtype with a contructor that does this automatically.